client/components/presales-zendesk-chat/index.tsx

```diff
@@ -1,27 +1,26 @@
+import { loadScript } from '@automattic/load-script';
 import { useEffect } from 'react';
 
 interface Props {
-	chatId: string | false;
+	chatKey: string | false;
 }
 
-const ZendeskChat = ( { chatId }: Props ) => {
+const ZendeskChat = ( { chatKey }: Props ) => {
 	useEffect( () => {
-		if ( ! chatId ) {
+		if ( ! chatKey ) {
 			return;
 		}
 
-		const script = document.createElement( 'script' );
-		script.src = 'https://static.zdassets.com/ekr/snippet.js?key=' + encodeURIComponent( chatId );
-		script.type = 'text/javascript';
-		script.id = 'ze-snippet';
+		loadScript(
+			'https://static.zdassets.com/ekr/snippet.js?key=' + encodeURIComponent( chatKey ),
+			() => {
+				return null;
+			},
```

Comment:
Why is that necessary? What would be the downside of `loadScript()` returning a promise in this case?

Reply:
> Why is that necessary? What would be the downside of `loadScript()` returning a promise in this case?

I might be wrong, but to pass the third argument to `loadScript` (optional params), we have to pass the 2nd callback arg, for which the type is defined as a function. And regardless of which function we pass, `loadScript` will not return a promise. 

I think the best I can do here is handle the error if it's returned, but I'm not sure it's worth it.
What do you think?

Reply:
Well, we could pass `undefined` as the second argument, no? In that case, it would return a promise but I don't see a problem with that TBH - let me know if I'm missing something.

And as for handling the error - ideally, we should make sure that if something goes wrong with the script, it doesn't take the entire app down. I believe we still have to wrap it in an error boundary - we do load the script to the `head`,  but if it fatals, it still should throw inside that same component tree. Would be worth testing it out just in case, of course. IIRC, last time we did something like this, an error boundary helped.

Reply:
> Well, we could pass undefined as the second argument, no?

That would require updating `loadScript`/`loadjQueryDependentScript` type definitions to accept that, but it makes sense - I'll do that. 

> Would be worth testing it out just in case

I'll try it!

Reply:
@tyxla So I tried to simulate an error in a script loaded via a script tag - but as far as I can tell, this won't directly affect the rest of the app. If we were then to interact with the JS we are loading (i.e, like we did with the Olark implementation), then it would make sense to add an error boundary. 

But in this case, we don't - the Zendesk JS does it's own thing, and errors within the loaded script shouldn't affect Calypso.